src/Illuminate/View/Concerns/CachesComponents.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Illuminate\View\Concerns;
+
+use Illuminate\Container\Container;
+use Illuminate\Contracts\Support\Htmlable;
+use Illuminate\Contracts\View\CanBeCached;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Cache;
+
+trait CachesComponents
+{
+    /**
+     * The attributes that can be passed to the component to control the
+     * caching behavior.
+     * @return array
+     */
+    protected function cacheAttributes()
+    {
+        return [
+            'x-cache',
+            'x-cache-key',
+            'x-cache-ttl',
+        ];
+    }
+
+    /**
+     * Determines if the component can be cached.
+     *
+     * @return bool
+     */
+    protected function canBeCached()
+    {
+        return $this instanceof CanBeCached;
+    }
+
+    /**
+     * Determines if the component should be cached.
+     *
+     * @return bool
+     */
+    protected function shouldBeCached()
+    {
+        return $this->attributes->get('x-cache', false) && $this->canBeCached();
+    }
+
+    /**
+     * The key used to cache the component.
+     *
+     * @param array $slots
```

Comment:
Please make sure all DocBlock parameters follow the coding standards of the rest of the framework.